controller/configentry_controller.go

```diff
@@ -210,12 +210,14 @@ func (r *ConfigEntryController) ReconcileEntry(
 	return ctrl.Result{}, nil
 }
 
-func (r *ConfigEntryController) consulNamespace(kubeNS string, isConsulNamespaced bool) string {
-	if isConsulNamespaced {
-		return namespaces.ConsulNamespace(kubeNS, r.EnableConsulNamespaces, r.ConsulDestinationNamespace, r.EnableNSMirroring, r.NSMirroringPrefix)
+func (r *ConfigEntryController) consulNamespace(namespace string, globalResource bool) string {
+	// Does not attempt to parse the namespace for global resources like ProxyDefaults or
+	// wildcard namespace destinations are they will not be prefixed and will remain "default"/"*".
+	if !globalResource && namespace != common.WildcardNamespace {
+		return namespaces.ConsulNamespace(namespace, r.EnableConsulNamespaces, r.ConsulDestinationNamespace, r.EnableNSMirroring, r.NSMirroringPrefix)
 	}
 	if r.EnableConsulNamespaces {
-		return common.DefaultConsulNamespace
+		return namespace
 	}
 	return ""
 }
```

Comment:
I tried to refactor this to make more sense. It's kinda confusing because a) we're passing in consulNS now as kubeNS and b) we have half of our namespace logic in this function and half in namespaces.ConsulNamespace.

I think maybe we need to put all of this logic into `namespaces.ConsulNamespace` but we can do that later.

Reply:
ohh i like the idea of moving it into consul namespace a lot. i kinda got tunnel visioned with the current implementation. this was honestly once of the trickiest things with service intentions. ill give it a fresh pair of eyes tomorrow and see if something strikes. If it doesn't, punt on it.